lms/djangoapps/bookmarks/serializers.py

```diff
@@ -0,0 +1,44 @@
+"""
+Serializers for Bookmarks.
+"""
+from rest_framework import serializers
+
+from .models import Bookmark
+
+
+class BookmarkSerializer(serializers.ModelSerializer):
+    """
+    Class that serializes the Bookmark model.
+    """
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'fields' arg up to the superclass
+        try:
+            fields = kwargs['context'].pop('fields', [])
+        except KeyError:
+            from .views import DEFAULT_FIELDS
+            fields = DEFAULT_FIELDS
+        # Instantiate the superclass normally
+        super(BookmarkSerializer, self).__init__(*args, **kwargs)
+
+        if fields:
```

Comment:
I'm not sure this "if" is correct. If you supply something falsey for fields (like []), then it means that you want all fields? It would make more sense to me to just always use the supplied list of "fields" (which will be the DEFAULT_FIELDS) if the user has not specified anything).